src/main/java/org/gameshop/management/GameManager.java

```diff
@@ -0,0 +1,122 @@
+package org.gameshop.management;
+
+import org.gameshop.database.dao.GameDao;
+import org.gameshop.database.dao.UserDao;
+import org.gameshop.model.Game;
+import org.gameshop.model.User;
+
+import java.sql.SQLException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Scanner;
+import lombok.RequiredArgsConstructor;
+
+
+@RequiredArgsConstructor
+public class GameManager {
+    private final Scanner scanner;
+    private final GameDao gameDao;
+    private final UserDao userDao;
+
+    public void showGameMenu() {
+        while (true) {
+            printGameMenu();
+            int choice = getUserChoice();
+            switch (choice) {
+                case 1:
+                    displayAllGames();
+                    break;
+                case 2:
+                    purchaseGame();
+                    break;
+                case 0:
+                    return;
+                default:
+                    System.out.println("Invalid choice. Please try again.");
+            }
+        }
+    }
+
+    private void printGameMenu() {
+        System.out.println("=== Game Management ===");
+        System.out.println("1. Display all games");
+        System.out.println("2. Purchase a game");
+        System.out.println("0. Go back to the main menu");
+    }
+
+    private int getUserChoice() {
+        System.out.print("Enter your choice: ");
+        while (!scanner.hasNextInt()) {
+            System.out.println("Invalid input. Please enter a number.");
+            System.out.print("Enter your choice: ");
+            scanner.next();
+        }
+        return scanner.nextInt();
+    }
+
+    private void displayAllGames() {
+        try {
+            List<Game> games = gameDao.getAllGames();
+            if (games.isEmpty()) {
+                System.out.println("No games available.");
+            } else {
+                System.out.println("List of available games:");
+                for (Game game : games) {
+                    System.out.println(gameToString(game));
+                }
+            }
+        } catch (SQLException e) {
+            System.out.println("Error displaying games: " + e.getMessage());
+        }
+    }
+
+    private void purchaseGame() {
+        System.out.println("Enter details to purchase a game:");
+
+        System.out.print("User ID: ");
```

Comment:
How does a user know his Id?In my opinion, the user should be authorized at this stage and then you can take his Id from object User

Reply:
Hi! Fixed it.